.github/workflows/cont_integration.yml

```diff
@@ -4,141 +4,137 @@ name: CI
-          args: --all-features --all-targets -- -D warnings
+      - name: checkout
+        uses: actions/checkout@v4
+      - name: Install Nix
+        uses: DeterminateSystems/nix-installer-action@v5
+      # Cache Nix artifacts
+      - uses: DeterminateSystems/magic-nix-cache-action@v2
+      # Cache Builds
+      - name: Build and Cache latest Rust
+        run: nix build -L . --keep-failed
+      # Tests: latest
+      - name: Test workspace all-features latest
+        run: nix build -L .#ci.latest.all --keep-failed
+      - name: Test workspace no-default-features latest
+        run: nix build -L .#ci.latest.noDefault --keep-failed
+      - name: Test bdk no-std latest
+        run: nix build -L .#ci.latest.noStdBdk --keep-failed
+      - name: Test bdk_chain no-std latest
+        run: nix build -L .#ci.latest.noStdChain --keep-failed
+      - name: Test bdk_esplora no-std latest
+        run: nix build -L .#ci.latest.noStdEsplora --keep-failed
+
+  test-MSRV:
+    name: Build, test, check MSRV
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+      - name: Install Nix
+        uses: DeterminateSystems/nix-installer-action@v5
+      # Cache Nix artifacts
+      - uses: DeterminateSystems/magic-nix-cache-action@v2
+      # Cache Builds
+      - name: Build and Cache MSRV
+        run: nix build -L .#MSRV --keep-failed
+      # Tests: MSRV
+      - name: Test workspace all-features MSRV
+        run: nix build -L .#ci.MSRV.all --keep-failed
+      - name: Test workspace no-default-features MSRV
+        run: nix build -L .#ci.MSRV.noDefault --keep-failed
+      - name: Test bdk no-std MSRV
+        run: nix build -L .#ci.MSRV.noStdBdk --keep-failed
+      - name: Test bdk_chain no-std MSRV
+        run: nix build -L .#ci.MSRV.noStdChain --keep-failed
+      - name: Test bdk_esplora no-std MSRV
+        run: nix build -L .#ci.MSRV.noStdEsplora --keep-failed
+
+  check-WASM:
+    name: Build and check WASM
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+      - name: Install Nix
+        uses: DeterminateSystems/nix-installer-action@v5
+      # Cache Nix artifacts
+      - uses: DeterminateSystems/magic-nix-cache-action@v2
+      # Cache Builds
+      - name: Build and Cache WASM
+        run: nix build -L .#WASM --keep-failed
+      # Checks: WASM
+      - name: Check bdk WASM
+        run: nix build -L .#ci.WASM.bdk --keep-failed
+      - name: Check bdk_esplora WASM
+        run: nix build -L .#ci.WASM.esplora --keep-failed
+
+  audit:
+    name: Audit based on rustsec/advisory-db
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+      - name: Install Nix
+        uses: DeterminateSystems/nix-installer-action@v5
+      # Cache Nix artifacts
+      - uses: DeterminateSystems/magic-nix-cache-action@v2
+      # Audit
+      - name: Audit
+        run: nix build -L .#ci.audit --keep-failed
```

Comment:
**suggestion:** Since the whole flake is locked, you probably want to run this one in a separate job and update the input before every run. This way it will detect latest advisories on every run.

https://github.com/fedimint/fedimint/blob/0ef5039086a4cd863484cbfacd638d8a0d42c233/.github/workflows/ci-nix.yml#L165

Reply:
I see.
However, this can change state depending when the user runs locally vs on CI.
I prefer to keep our `flake.lock` in the same state for CI and local.
We also have a job that updates the flake every week on CI.